app/models/delay_henka/scheduled_change.rb

```diff
@@ -11,18 +11,20 @@ class ScheduledChange < ApplicationRecord
     belongs_to :changeable, polymorphic: true
 
     validates :submitted_by_id, :attribute_name, presence: true
+    validates :schedule_at, presence: true, on: :update
     validates :state, inclusion: { in: STATES.values }
+    before_create :set_default_schedule_at
     after_initialize :set_initial_state, if: :new_record?
 
     scope :staged, -> { where(state: STATES[:STAGED]) }
 
-    def self.schedule(record:, changes:, by_id:)
+    def self.schedule(record:, changes:, by_id:, schedule_at: nil)
```

Comment:
If you set Time.current as default here, you wouldn’t need a separate callback to set it